Raft elects two leaders at the same term

**Describe the bug**

There are some scenarios where raft elects two leaders at the same time.  

In the following log 0 voted for 1 and later for 2 at term 1195
```
07:51:16.133 [] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Accepted VoteRequest{term=1195, candidate=1, lastLogIndex=481, lastLogTerm=1129}: candidate's log is up-to-date
07:51:16.133 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote 1
07:51:16.134 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Voted for 1
07:51:16.134 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.134 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending VoteResponse{status=OK, term=1195, voted=true}

...

07:51:16.138 [] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Accepted VoteRequest{term=1195, candidate=2, lastLogIndex=750, lastLogTerm=1189}: candidate's log is up-to-date
07:51:16.138 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote 2
07:51:16.139 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Voted for 2
07:51:16.139 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.139 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending VoteResponse{status=OK, term=1195, voted=true}

```

**To Reproduce**

It can be reproduced by running `io.atomix.raft.RandomizedRaftTest` in branch https://github.com/zeebe-io/zeebe/tree/dd-test-raft

**Expected behavior**
There shouldn't be two leaders in a term.

**Log/Stacktrace**
If possible add the full stacktrace or Zeebe log which contains the issue. 

<details><summary>Full Stacktrace</summary>
 <p>

```
07:51:16.133 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Received VoteRequest{term=1195, candidate=1, lastLogIndex=481, lastLogTerm=1129}
07:51:16.133 [] TRACE io.atomix.raft.storage.system.MetaStore - Store term 1195
07:51:16.133 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.133 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Set term 1195
07:51:16.133 [] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Accepted VoteRequest{term=1195, candidate=1, lastLogIndex=481, lastLogTerm=1129}: candidate's log is up-to-date
07:51:16.133 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote 1
07:51:16.134 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Voted for 1
07:51:16.134 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.134 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending VoteResponse{status=OK, term=1195, voted=true}
07:51:16.134 [] INFO  TEST - Running deliverAllMessage to 1 on 1
07:51:16.134 [] INFO  TEST - Running runNextTask on 2
07:51:16.134 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{2-partition-1}{role=FOLLOWER} - Received rejected poll from DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.453Z}
07:51:16.134 [] INFO  TEST - Running clientAppend on 2
07:51:16.134 [] INFO  TEST - Running runNextTask on 2
07:51:16.134 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{2-partition-1}{role=FOLLOWER} - Received accepted poll from DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.453Z}
07:51:16.134 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{2-partition-1} - Transitioning to CANDIDATE
07:51:16.134 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{2-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.134 [] INFO  io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Starting election
07:51:16.134 [] TRACE io.atomix.raft.storage.system.MetaStore - Store term 1195
07:51:16.135 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.135 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{2-partition-1} - Set term 1195
07:51:16.135 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote 2
07:51:16.135 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{2-partition-1} - Voted for 2
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Requesting votes for term 1195
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Requesting vote from DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.453Z} for term 1195
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Requesting vote from DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.453Z} for term 1195
07:51:16.135 [] INFO  TEST - Running clientAppendOnLeader on 2
07:51:16.135 [] INFO  TEST - Running tickHeartBeatTimeout on 0
07:51:16.135 [] INFO  TEST - Running runNextTask on 2
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Received VoteRequest{term=1195, candidate=1, lastLogIndex=481, lastLogTerm=1129}
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Sending VoteResponse{status=OK, term=1195, voted=false}
07:51:16.135 [] INFO  TEST - Running runNextTask on 1
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{1-partition-1}{role=CANDIDATE} - Received successful vote from DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.451Z}
07:51:16.135 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{1-partition-1} - Transitioning to LEADER
07:51:16.135 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{1-partition-1}{role=CANDIDATE} - Cancelling election
07:51:16.135 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{1-partition-1} - Found leader 1
07:51:16.135 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.136 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Appended Indexed{index=482, entry=InitializeEntry{term=1195, timestamp=2020-09-16 07:51:16,135}}
07:51:16.136 [] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Configuring 2
07:51:16.136 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Sending ConfigureRequest{term=1195, leader=1, index=0, timestamp=1600235467451, members=[DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}]} to 2
07:51:16.136 [] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Configuring 0
07:51:16.136 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Sending ConfigureRequest{term=1195, leader=1, index=0, timestamp=1600235467451, members=[DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}]} to 0
07:51:16.136 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Starting append timer on fix rate of PT0.15S
07:51:16.136 [] INFO  TEST - Running processNextMessage on 0
07:51:16.136 [] INFO  TEST - Running deliverAllMessage to 1 on 1
07:51:16.136 [] INFO  TEST - Running deliverAllMessage to 0 on 1
07:51:16.136 [] INFO  TEST - Running clientAppend on 0
07:51:16.136 [] INFO  TEST - Running runNextTask on 0
07:51:16.136 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Received ConfigureRequest{term=1195, leader=1, index=0, timestamp=1600235467451, members=[DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}, DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.451050Z}]}
07:51:16.136 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Found leader 1
07:51:16.136 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.137 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending ConfigureResponse{status=OK}
07:51:16.137 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.137 [] INFO  TEST - Running deliverAllMessage to 2 on 2
07:51:16.137 [] INFO  TEST - Running dropNextMessage to 1 on 0
07:51:16.137 [] INFO  TEST - Running processAllMessage on 0
07:51:16.137 [] INFO  TEST - Running tickElectionTimeout on 0
07:51:16.137 [] INFO  TEST - Running tickElectionTimeout on 0
07:51:16.137 [] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - No heartbeat from DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.449Z} in the last PT0.873S (calculated from last 0 ms), sending poll requests
07:51:16.137 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Polling members [DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.449Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.449Z}]
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Polling DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.449Z} for next term 1196
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Polling DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.449Z} for next term 1196
07:51:16.138 [] INFO  TEST - Running deliverAllMessage to 2 on 0
07:51:16.138 [] INFO  TEST - Running runNextTask on 0
07:51:16.138 [] INFO  TEST - Running deliverAllMessage to 1 on 2
07:51:16.138 [] INFO  TEST - Running dropNextMessage to 1 on 2
07:51:16.138 [] INFO  TEST - Running processNextMessage on 1
07:51:16.138 [] INFO  TEST - Running clientAppend on 0
07:51:16.138 [] INFO  TEST - Running clientAppendOnLeader on 1
07:51:16.138 [] INFO  TEST - Running clientAppendOnLeader on 2
07:51:16.138 [] INFO  TEST - Running tickElectionTimeout on 0
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Failed to poll a majority of the cluster in PT0.5S
07:51:16.138 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.138 [] INFO  TEST - Running tickHeartBeatTimeout on 0
07:51:16.138 [] INFO  TEST - Running clientAppendOnLeader on 0
07:51:16.138 [] INFO  TEST - Running tickElectionTimeout on 1
07:51:16.138 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Sending AppendRequest{term=1195, leader=1, prevLogIndex=481, prevLogTerm=1129, entries=0, commitIndex=42} to 2
07:51:16.138 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Sending AppendRequest{term=1195, leader=1, prevLogIndex=481, prevLogTerm=1129, entries=0, commitIndex=42} to 0
07:51:16.138 [] INFO  TEST - Running deliverAllMessage to 0 on 2
07:51:16.138 [] INFO  TEST - Running deliverNextMessage to 1 on 0
07:51:16.138 [] INFO  TEST - Running clientAppend on 1
07:51:16.138 [] INFO  TEST - Running deliverNextMessage to 2 on 0
07:51:16.138 [] INFO  TEST - Running runNextTask on 0
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Received PollRequest{term=1194, candidate=2, lastLogIndex=750, lastLogTerm=1189}
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Rejected PollRequest{term=1194, candidate=2, lastLogIndex=750, lastLogTerm=1189}: candidate's term is less than the current term
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending PollResponse{status=OK, term=1195, accepted=false}
07:51:16.138 [] INFO  TEST - Running clientAppend on 2
07:51:16.138 [] INFO  TEST - Running tickHeartBeatTimeout on 0
07:51:16.138 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Received VoteRequest{term=1195, candidate=2, lastLogIndex=750, lastLogTerm=1189}
07:51:16.138 [] INFO  io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Accepted VoteRequest{term=1195, candidate=2, lastLogIndex=750, lastLogTerm=1189}: candidate's log is up-to-date
07:51:16.138 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote 2
07:51:16.139 [] DEBUG io.atomix.raft.impl.RaftContext - RaftServer{0-partition-1} - Voted for 2
07:51:16.139 [] TRACE io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Cancelling heartbeat timer
07:51:16.139 [] DEBUG io.atomix.raft.roles.FollowerRole - RaftServer{0-partition-1}{role=FOLLOWER} - Sending VoteResponse{status=OK, term=1195, voted=true}

07:51:16.139 [] INFO  TEST - Running deliverNextMessage to 2 on 0
07:51:16.139 [] INFO  TEST - Running deliverNextMessage to 1 on 2
07:51:16.139 [] INFO  TEST - Running runNextTask on 1
07:51:16.139 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Appended Indexed{index=483, entry=ZeebeEntry{term=1195, timestamp=1600235476139, lowestPosition=0, highestPosition=1}}
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 1 on 1
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 0 on 0
07:51:16.139 [] INFO  TEST - Running processAllMessage on 2
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 1 on 1
07:51:16.139 [] INFO  TEST - Running clientAppendOnLeader on 1
07:51:16.139 [] INFO  TEST - Running dropNextMessage to 1 on 0
07:51:16.139 [] INFO  TEST - Running dropNextMessage to 1 on 1
07:51:16.139 [] INFO  TEST - Running processNextMessage on 1
07:51:16.139 [] INFO  TEST - Running dropNextMessage to 2 on 2
07:51:16.139 [] INFO  TEST - Running runNextTask on 1
07:51:16.139 [] DEBUG io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Received VoteRequest{term=1195, candidate=2, lastLogIndex=750, lastLogTerm=1189}
07:51:16.139 [] DEBUG io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Sending VoteResponse{status=OK, term=1195, voted=false}
07:51:16.139 [] INFO  TEST - Running runNextTask on 1
07:51:16.139 [] DEBUG io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Received PollRequest{term=1195, candidate=0, lastLogIndex=105, lastLogTerm=161}
07:51:16.139 [] DEBUG io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Sending PollResponse{status=OK, term=1195, accepted=false}
07:51:16.139 [] INFO  TEST - Running deliverNextMessage to 2 on 2
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 2 on 1
07:51:16.139 [] INFO  TEST - Running tickElectionTimeout on 1
07:51:16.139 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{1-partition-1} - Received ConfigureResponse{status=OK} from 0
07:51:16.139 [] INFO  TEST - Running runNextTask on 1
07:51:16.139 [] INFO  TEST - Running tickElectionTimeout on 2
07:51:16.139 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Received PollRequest{term=1195, candidate=0, lastLogIndex=105, lastLogTerm=161}
07:51:16.139 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Rejected PollRequest{term=1195, candidate=0, lastLogIndex=105, lastLogTerm=161}: candidate's last log entry (161) is at a lower term than the local log (1189)
07:51:16.139 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Sending PollResponse{status=OK, term=1195, accepted=false}
07:51:16.139 [] INFO  TEST - Running clientAppend on 0
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 2 on 0
07:51:16.139 [] INFO  TEST - Running clientAppend on 1
07:51:16.139 [] INFO  TEST - Running clientAppend on 2
07:51:16.139 [] INFO  TEST - Running deliverNextMessage to 0 on 0
07:51:16.139 [] INFO  TEST - Running clientAppendOnLeader on 0
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 2 on 0
07:51:16.139 [] INFO  TEST - Running processNextMessage on 1
07:51:16.139 [] INFO  TEST - Running deliverNextMessage to 0 on 0
07:51:16.139 [] INFO  TEST - Running tickHeartBeatTimeout on 2
07:51:16.139 [] INFO  TEST - Running tickHeartBeatTimeout on 1
07:51:16.139 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{1-partition-1}{role=LEADER} - Appended Indexed{index=484, entry=ZeebeEntry{term=1195, timestamp=1600235476139, lowestPosition=0, highestPosition=1}}
07:51:16.139 [] INFO  TEST - Running clientAppendOnLeader on 1
07:51:16.139 [] INFO  TEST - Running tick 50ms on 1
07:51:16.139 [] INFO  TEST - Running clientAppend on 1
07:51:16.139 [] INFO  TEST - Running deliverAllMessage to 2 on 1
07:51:16.139 [] INFO  TEST - Running processNextMessage on 2
07:51:16.139 [] INFO  TEST - Running runNextTask on 2
07:51:16.139 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Received successful vote from DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.453Z}
07:51:16.139 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{2-partition-1} - Transitioning to LEADER
07:51:16.139 [] DEBUG io.atomix.raft.roles.CandidateRole - RaftServer{2-partition-1}{role=CANDIDATE} - Cancelling election
07:51:16.139 [] INFO  io.atomix.raft.impl.RaftContext - RaftServer{2-partition-1} - Found leader 2
07:51:16.139 [] TRACE io.atomix.raft.storage.system.MetaStore - Store vote null
07:51:16.140 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{2-partition-1}{role=LEADER} - Appended Indexed{index=751, entry=InitializeEntry{term=1195, timestamp=2020-09-16 07:51:16,139}}
07:51:16.140 [] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{2-partition-1} - Configuring 1
07:51:16.140 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{2-partition-1} - Sending ConfigureRequest{term=1195, leader=2, index=0, timestamp=1600235467453, members=[DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}, DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}]} to 1
07:51:16.140 [] DEBUG io.atomix.raft.roles.LeaderAppender - RaftServer{2-partition-1} - Configuring 0
07:51:16.140 [] TRACE io.atomix.raft.roles.LeaderAppender - RaftServer{2-partition-1} - Sending ConfigureRequest{term=1195, leader=2, index=0, timestamp=1600235467453, members=[DefaultRaftMember{id=2, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}, DefaultRaftMember{id=1, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}, DefaultRaftMember{id=0, type=ACTIVE, updated=2020-09-16T05:51:07.453279Z}]} to 0
07:51:16.140 [] TRACE io.atomix.raft.roles.LeaderRole - RaftServer{2-partition-1}{role=LEADER} - Starting append timer on fix rate of PT0.15S
07:51:16.141 [] DEBUG io.atomix.raft.impl.zeebe.LogCompactor - LogCompactor{1-partition-1} - Closing the log compactor 1-partition-1
07:51:16.141 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=1, version=1, index=1}
07:51:16.141 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=2, version=1, index=320}
07:51:16.141 [] DEBUG io.atomix.raft.impl.zeebe.LogCompactor - LogCompactor{2-partition-1} - Closing the log compactor 2-partition-1
07:51:16.141 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=1, version=1, index=1}
07:51:16.141 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=2, version=1, index=315}
07:51:16.141 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=3, version=1, index=626}
07:51:16.142 [] DEBUG io.atomix.raft.impl.zeebe.LogCompactor - LogCompactor{0-partition-1} - Closing the log compactor 0-partition-1
07:51:16.142 [] DEBUG io.atomix.storage.journal.SegmentedJournal - Closing segment: JournalSegment{id=1, version=1, index=1}

java.lang.AssertionError: Found two leaders 1 2 at term 1195

	at io.atomix.raft.RaftContextRule.updateAndVerifyLeaderTerm(RaftContextRule.java:332)
	at io.atomix.raft.RaftContextRule.lambda$assertOnlyOneLeader$11(RaftContextRule.java:321)
	at java.base/java.util.HashMap$Values.forEach(HashMap.java:976)
	at io.atomix.raft.RaftContextRule.assertOnlyOneLeader(RaftContextRule.java:321)
	at io.atomix.raft.RandomizedRaftTest.verifyRaftProperties(RandomizedRaftTest.java:85)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)


```

</p>
</details>

--- a/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java
+++ b/atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java
@@ -954,8 +954,7 @@ public class RaftContext implements AutoCloseable {
         }
       }
 
-      lastVotedFor = null;
-      meta.storeVote(null);
+      log.trace("Set leader {}", this.leader);
     }
   }
 

